### ISSUE TITLE
Ubuntu 18.04, steamshovel does not start due to pybinding issue? (Trac #2378)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2370]
Hi,

on current combo/trunk on an Ubuntu 18.05 (nspawn container with linked --bind-ro=/tmp/.X11-unix and exported DISPLAY variable in case this is important), after only making steamshovel with make -j 8 steamshovel, it does not start complaining about the wrong python version in the pybindings:

```text
 root@ubuntu:~/combo_trunk/build# make -j 8 steamshovel
 root@ubuntu:~/combo_trunk/build# ./env-shell.sh 
 ************************************************************************
 *                                                                      *
 *                   W E L C O M E  to  I C E T R A Y                   *
 *                                                                      *
 *                   Version combo.trunk     r177798                    *
 *                                                                      *
 *                You are welcome to visit our Web site                 *
 *                        http://icecube.umd.edu                        *
 *                                                                      *
 ************************************************************************ 

 Icetray environment has:
   I3_SRC       = /root/combo_trunk/src
   I3_BUILD     = /root/combo_trunk/build
   I3_TESTDATA should be set to an existing directory path
   Python       = 2.7.15

 root@ubuntu:~/combo_trunk/build# steamshovel
 ERROR (steamshovel): Problem loading icecube python bindings:
 Traceback (most recent call last):
  File "/root/combo_trunk/build/lib/icecube/dataio/__init__.py", line 1, in <module>
    from icecube import icetray, dataclasses
  File "/root/combo_trunk/build/lib/icecube/icetray/__init__.py", line 1, in <module>
    from icecube.load_pybindings import load_pybindings
  File "/root/combo_trunk/build/lib/icecube/load_pybindings.py", line 31, in <module>
    .format(PYTHON_VERSION,platform.python_version()))
 Exception: IceTray was compiled with version 2.7.15 but is curently being run with  2.7.15+
 Are there unbuilt pybindings? (embed.cpp:49 in  scripting::PyInterpreter::PyInterpreter(char*))
 Segmentation fault (core dumped)
```

Running a full make atm to see if this fixes it but even then we should be able to build individual projects.

Migrated from https://code.icecube.wisc.edu/ticket/2378
```json
{
    "status": "closed", 
    "changetime": "2019-12-11T15:30:18", 
    "description": "Hi,\n\non current combo/trunk on an Ubuntu 18.05 (nspawn container with linked --bind-ro=/tmp/.X11-unix and exported DISPLAY variable in case this is important), after only making steamshovel with make -j 8 steamshovel, it does not start complaining about the wrong python version in the pybindings:\n\n{{{\n root@ubuntu:~/combo_trunk/build# make -j 8 steamshovel\n root@ubuntu:~/combo_trunk/build# ./env-shell.sh \n ************************************************************************\n *                                                                      *\n *                   W E L C O M E  to  I C E T R A Y                   *\n *                                                                      *\n *                   Version combo.trunk     r177798                    *\n *                                                                      *\n *                You are welcome to visit our Web site                 *\n *                        http://icecube.umd.edu                        *\n *                                                                      *\n ************************************************************************ \n\n Icetray environment has:\n   I3_SRC       = /root/combo_trunk/src\n   I3_BUILD     = /root/combo_trunk/build\n   I3_TESTDATA should be set to an existing directory path\n   Python       = 2.7.15\n\n root@ubuntu:~/combo_trunk/build# steamshovel\n ERROR (steamshovel): Problem loading icecube python bindings:\n Traceback (most recent call last):\n  File \"/root/combo_trunk/build/lib/icecube/dataio/__init__.py\", line 1, in <module>\n    from icecube import icetray, dataclasses\n  File \"/root/combo_trunk/build/lib/icecube/icetray/__init__.py\", line 1, in <module>\n    from icecube.load_pybindings import load_pybindings\n  File \"/root/combo_trunk/build/lib/icecube/load_pybindings.py\", line 31, in <module>\n    .format(PYTHON_VERSION,platform.python_version()))\n Exception: IceTray was compiled with version 2.7.15 but is curently being run with  2.7.15+\n Are there unbuilt pybindings? (embed.cpp:49 in  scripting::PyInterpreter::PyInterpreter(char*))\n Segmentation fault (core dumped)\n}}}\n\nRunning a full make atm to see if this fixes it but even then we should be able to build individual projects.", 
    "reporter": "flauber", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1576078218546095", 
    "component": "combo reconstruction", 
    "summary": "Ubuntu 18.04, steamshovel does not start due to pybinding issue?", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2019-12-10T09:30:16", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20191210T16:59:50: 
* flauber commented:

> So the issue is not with the container as I was able to build the hole thing with combo/stable but seems to be with combo/trunk.
> 
> One thing I noticed is that stable picks up python3 while trunk picks up python2 on the same container. Was there maybe a change to the cmake detection which broke this?


* flauber changed summary from "Ubuntu 18.05, steamshovel does not start due to pybinding issue?" to "Ubuntu 18.04, steamshovel does not start due to pybinding issue?"

**Comment by icecube-trac** [comment]
Trac update at 20191210T17:15:11: david.schultz commented:

> Yes, there was some churn in the cmake detection of python.  The one in trunk is better, but hasn't migrated to stable yet.

**Comment by icecube-trac** [comment]
Trac update at 20191210T17:16:02: olivas commented:

> https://code.icecube.wisc.edu/projects/icecube/changeset/177705/IceCube/meta-projects/combo/trunk
> 
>

**Comment by icecube-trac** [comment]
Trac update at 20191210T17:17:27: 
* olivas commented:

> We'll get this fixed.  I've seen that on a couple bots as well.


* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20191210T17:30:29: flauber commented:

> @davidschulz
> 
> Are you sure? Stable works and pick sup Pyhton3 as intended but trunk doe snot work and pick up python2. I would have thought that the detection in stable is better in that case. Or is the trunk one in other regards better?

**Comment by icecube-trac** [comment]
Trac update at 20191211T15:30:18: 
* olivas changed resolution from "" to "fixed"
* olivas commented:

> In changeset 177830:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="177830"
> only compare the numeric parts of the python versions, since cmake and python don't report the same version numbers. fixes #2370 hopefully.
> ```


* olivas changed status from "assigned" to "closed"
